### PR TITLE
Nacl for ext pillar [WIP]

### PR DESCRIPTION
**Pull request by epcim**
### What does this PR do?
Decrypt NACL strings on ext_pillars. (we already have this for gpg)

Note: this PR depends on nacl renderer, once merged: 
- https://github.com/saltstack/salt/pull/41868

(might be related:  https://github.com/saltstack/salt/pull/41956)

I count to rebase upstream/develp once above two merged.

### Tests written?
No

### CODE REVIEW

**Comment by garethgreenaway** [comment]
re-run ubuntu

File: salt/pillar/nacl.py [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    ext_pillar:
+      - stack: /path/to/stack.cfg
+      - nacl: {}
+
+Set ``nacl.config`` in your config.
+
+'''
+
+from __future__ import absolute_import
+import salt.utils
```
Comment (cachedout) [inline salt/pillar/nacl.py]: Why is this import necessary? I'm not seeing it used anywhere. I'm guessing this was copied from the GPG rendered but tbh I'm not sure I follow the logic there either.

Comment (cachedout) [reply salt/pillar/nacl.py]: @epcim Did you see the above question?

**Comment by cachedout** [comment]
Go Go Jenkins!

File: salt/pillar/nacl.py (hunk 1 above)
Comment (epcim) [reply salt/pillar/nacl.py]: salt.utils is quite sure not needed, it's copy/paste from other recently added ext pillar processors.

**Comment by epcim** [comment]
Is there anything wrong with testing? I don't see failures.

**Comment by rallytime** [comment]
HI @epcim - the pylint failures are listed here:

https://jenkins.saltstack.com/job/PR/job/salt-pr-lint-n/14189/violations/file/salt/pillar/nacl.py/

**Review by cachedout — APPROVED** [review]
